Tests/Common/Communication/EventStorageTests.swift

```diff
@@ -0,0 +1,165 @@
+@testable import CioInternalCommon
+import SharedTests
+import XCTest
+
+class EventStorageTest: UnitTest {
+    var eventStorageManager: EventStorageManager!
+
+    override func setUp() {
+        super.setUp()
+        if let tempDirectory = createTemporaryDirectory() {
+            eventStorageManager = EventStorageManager(logger: log, jsonAdapter: jsonAdapter)
+            Task {
+                await eventStorageManager.updateBaseDirectory(baseDirectory: tempDirectory)
```

Comment:
A better option would be to make the setup function async: `override func setUp() async`

Reply:
I am not sure I fully get it, we are overriding the 
```
/*!
 * @method -setUp
 * Setup method called before the invocation of each test method in the class.
 */
- (void)setUp;
```
from XCAbstractTest, there isn't an async one that we can use. We can probably create another async function, and then call that from this method inside `Task`, but I am not sure how if that would be better or same.